pkg/networkserver/redis/registry.go

```diff
@@ -58,6 +59,8 @@ const (
 type DeviceRegistry struct {
 	Redis   *ttnredis.Client
 	LockTTL time.Duration
+	// CompatibilityVersion denotes the lowest possible stack version the registry should be compatible with.
+	CompatibiltyVersion semver.Version
```

Comment:
Where is this set?

Reply:
See OP:
> At that point I will add the configuration toconfig as well - for now there's no use for it. (this could only be used on new deployments with empty registries and we don't plan that any time soon)

This will be set when there's a reason to do so.

Reply:
While indeed we won't indeed deploy a new cluster too soon, there are customers that run the enterprise stack themselves (or that can pickup the AMI for the Marketplace) which will start on a clean registry. So there is a need for it.

Reply:
Ok, but then we have to discuss this, because this is something we probably want to have shared between components, maybe not - adding this to the config now means that we cannot change it later and I'm not very confident to do this just yet. We don't even know if this is the right approach to handling this issue.
So what I'm saying is let's file an issue, discuss and then implement accordingly.
This PR addresses the fact that 3.10 is currently broken and so is a bugfix. I do not want to add new features here.